processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -159,8 +159,18 @@ public GroupByQueryQueryToolChest(
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
       final List<AggregatorFactory> aggs = Lists.newArrayList();
+
       for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
```

Comment:
I tested a few cases with multiple different inner / outer aggregator types and didn't see any issues. It seems like we're just using the innerQuery to build an intermediate incremental index in such a way that it contains the columns expected by the outerQuery which we then execute on the intermediate index.. but yeah, it was a bit weird which is why I'm not too sure about the fix.

Can you come up with a query that fails? For reference, here is the query I was using that exercises different combinations of aggregators:

```
{
  "queryType": "groupBy",
  "dataSource": {
    "type": "query",
    "query": {
      "queryType": "groupBy",
      "dataSource": "wikipedia",
      "granularity": "all",
      "dimensions": [
        "language",
        "page"
      ],
      "aggregations": [
        {
          "type": "longSum",
          "fieldName": "count",
          "name": "edit_count"
        },
        {
          "type": "doubleSum",
          "fieldName": "count",
          "name": "edit_count_2"
        },
        {
          "type": "javascript",
          "name": "js_inner",
          "fieldNames": [
            "count"
          ],
          "fnAggregate": "function(current, v) { return current + 4*v; } ",
          "fnCombine": "function(partialA, partialB) { return partialA + partialB; }",
          "fnReset": "function() { return 0; }"
        }
      ],
      "intervals": [
        "2012-01-01T00:00:00.000\/2020-01-03T00:00:00.000"
      ]
    }
  },
  "granularity": "hour",
  "dimensions": [
    "language"
  ],
  "aggregations": [
    {
      "type": "longSum",
      "fieldName": "edit_count",
      "name": "longSum_1"
    },
    {
      "type": "longSum",
      "fieldName": "edit_count_2",
      "name": "longSum_2"
    },
    {
      "type": "doubleSum",
      "fieldName": "edit_count",
      "name": "doubleSum_1"
    },
    {
      "type": "longSum",
      "fieldName": "js_inner",
      "name": "longSum_3"
    },
    {
      "type": "javascript",
      "name": "js_1",
      "fieldNames": [
        "edit_count"
      ],
      "fnAggregate": "function(current, v) { return current + 2*v; } ",
      "fnCombine": "function(partialA, partialB) { return partialA + partialB; }",
      "fnReset": "function() { return 0; }"
    },
    {
      "type": "javascript",
      "name": "js_2",
      "fieldNames": [
        "edit_count"
      ],
      "fnAggregate": "function(current, v) { return current + 3*v; } ",
      "fnCombine": "function(partialA, partialB) { return partialA + partialB; }",
      "fnReset": "function() { return 0; }"
    }
  ],
  "intervals": [
    "2012-01-01T00:00:00.000\/2020-01-03T00:00:00.000"
  ]
}
```